tests/PathMappingRepositoryTest.php

```diff
@@ -17,71 +17,30 @@
 use Puli\Repository\Resource\DirectoryResource;
 use Puli\Repository\Resource\FileResource;
 use Puli\Repository\Resource\GenericResource;
-use Puli\Repository\Tests\Resource\TestFilesystemDirectory;
-use Puli\Repository\Tests\Resource\TestFilesystemFile;
-use Symfony\Component\Filesystem\Filesystem;
-use Webmozart\Glob\Test\TestUtil;
+use Webmozart\KeyValueStore\Api\KeyValueStore;
 use Webmozart\KeyValueStore\ArrayStore;
 
 /**
- * @since  1.0
- *
  * @author Bernhard Schussek <bschussek@gmail.com>
  * @author Titouan Galopin <galopintitouan@gmail.com>
  */
-class PathMappingRepositoryTest extends AbstractEditableRepositoryTest
+class PathMappingRepositoryTest extends AbstractPathMappingRepositoryTest
 {
     /**
-     * @var ArrayStore
-     */
-    protected $store;
-
-    /**
-     * @var PathMappingRepository
-     */
-    protected $repo;
-
-    /**
-     * Temporary directory for test files.
-     *
-     * @var string
-     */
-    protected $tempDir;
-
-    /**
      * Counter to avoid collisions during tests on directories.
      *
      * @var int
      */
     protected static $createdDirectories = 0;
 
-    /**
-     * Counter to avoid collisions during tests on files.
-     *
-     * @var int
-     */
-    protected static $createdFiles = 0;
-
-    protected function setUp()
+    protected function createBaseDirectoryRepository(KeyValueStore $store, $baseDirectory)
     {
-        parent::setUp();
-
-        $this->tempDir = TestUtil::makeTempDir('puli-repository', __CLASS__);
-        $this->store = new ArrayStore();
-        $this->repo = new PathMappingRepository($this->store);
-    }
-
-    protected function tearDown()
-    {
-        parent::tearDown();
-
-        $filesystem = new Filesystem();
-        $filesystem->remove($this->tempDir);
+        return new PathMappingRepository($store, $baseDirectory);
     }
 
     protected function createPrefilledRepository(Resource $root)
     {
-        $repo = new PathMappingRepository(new ArrayStore());
+        $repo = new PathMappingRepository(new ArrayStore(), ('\\' === DIRECTORY_SEPARATOR) ? 'C:/' : '/');
```

Comment:
Can we guarantee that this is always `C:/`? This smells like a volatile test to me.